third-party/realdds/src/dds-device-broadcaster.cpp

```diff
@@ -132,17 +141,42 @@ dds_device_broadcaster::dds_device_broadcaster( std::shared_ptr< dds_publisher >
     _manager = participant_broadcast_manager[participant_guid].instance( publisher );
     _writer = _manager->register_broadcaster( this );
 
-    broadcast();  // possible we have no subscribers, but can't hurt
+    _manager->broadcast();  // possible we have no subscribers, but can't hurt
 }
 
 
 void dds_device_broadcaster::broadcast() const
 {
+    // Called from the manager, in its worker thread, whenever our device-info needs broadcasting (which can happen
+    // multiple times, once per new participant we see). This happens automatically when a dds_device_broadcaster is
+    // instantiated.
     try
     {
         topics::flexible_msg msg( _device_info.to_json() );
-        LOG_DEBUG( "sending device-info message " << slice( msg.custom_data< char const >(), msg._data.size() ) );
+        LOG_DEBUG( "[" << _device_info.debug_name() << "] broadcasting device-info " << _device_info.to_json() );
         std::move( msg ).write_to( *_writer );
+
+        // If a broadcast callback is asked for, we wait for acks and call it on the first broadcast (and never again)
+        if( _on_ack )
+        {
+            std::thread(
+                [on_ack = std::move( _on_ack ),
```

Comment:
Why do we move `_on_ack`? What will happen in the next broadcast?

Reply:
The next broadcast is automatic, when another participant joins. We only need the on-ack callback after we specifically create the broadcaster, to do something.

Reply:
(see the comment above -- we only want to call the on-ack the first time)